cms/models/pagemodel.py

```diff
@@ -902,32 +902,34 @@ def get_redirect(self, language=None, fallback=True, force_reload=False):
         return self.get_page_content_obj_attribute("redirect", language, fallback, force_reload)
 
     def _get_title_cache(self, language, fallback, force_reload):
-        if not language:
-            language = get_language()
-
-        force_reload = (force_reload or language not in self.title_cache)
-
-        if fallback and not self.title_cache.get(language):
-            # language can be in the cache but might be an EmptyPageContent instance
+        def get_fallback_language(page, language):
             fallback_langs = i18n.get_fallback_languages(language)
             for lang in fallback_langs:
-                if self.title_cache.get(lang):
+                if page.title_cache.get(lang):
                     return lang
 
+        if not language:
+            language = get_language()
+
```

Comment:
You are correct, you'll see the effects of this in the test that I provided. If you ran the test prior to this patch the test would fail. With this patch, the test passes.